### DISCUSSION TITLE
About AVX compatibility and hardware acceleration

### DISCUSSION THREAD

**Discussion by matheus-j-ressurreicao** [GrandOrgue/grandorgue#932] [Developers] [answered]
Hello GrandOrgue development team. I would very much like to open a discussion (actually it would be more of a question) as to whether there is already a compatibility in the application code that provides support for AVX instructions when using the CPU. If there is no such compatibility, I would very much like to ask if this would have any relevant benefits for the long-term development planning of GrandOrgue and even thinking about cross-platform compatibility? Also, I would like to ask if the application supports hardware acceleration in any way, even in the sense that e.g. CPU and RAM work together for a better overall functioning of the application? I very much hope that this question makes some technical sense! Thanks!

**Comment by larspalo** [comment, marked as answer]
Please, understand that GO does not operate directly against the CPU with any hard coded assembler code base (anymore). It uses whatever the operating system provide, as well as what the (higher) intermediate layers of c++ (compiler etc), wxwidgets, RtAudio, RtMidi, PortAudio, etc provides. With most modern operating systems available, you will have access to hardware acceleration - no worries there. To maximize the cross-platform compatibility there won't be any specific instructions for any single architecture in the GO code (as you can see when you read the code), instead we will rely upon the compiler to provide that adaptation for us.

What you as the user can do is specify at what concurrency level you want the sound engine of GO to work, and tailor the sound output to work as best as your specific hardware can perform. Besides that, the general rules of computer usage applies... Have as little other processes using the resources of the computer as possible to allow GO to use your computer to its maximum capacity.

**Comment by matheus-j-ressurreicao** [comment]
@larspalo thank you very much for the clarification. The reason for this question is in fact to understand whether GrandOrgue is currently making the best possible use of the computational resources (processing, RAM, disk reading speed, etc.) for its purpose. Thank you very much for clarifying that today the application already supports hardware acceleration and makes use of it for the best performance. Looking forward to future releases!
